packages/block/src/types.ts

```diff
@@ -101,6 +101,10 @@ export interface BlockData {
   uncleHeaders?: Array<HeaderData>
 }
 
+export interface HeaderCache {
+  hash: Buffer | undefined
+}
```

Comment:
One small nit: please don't export this here but only define interally in the class itself (should be possible, right?). Otherwise this is getting part of the public part of the API which is not what we want.

Reply:
Ah ok, I don't think I had made that connection before. PR updated. So to confirm my understanding, `types.ts` in the various packages is only for things that are part of the public-facing API of that package?

Reply:
Yes, that's correct. People use these types for typing their applications and the more "garbage" we add there, the more confusing it will be (to be fair: we are generally still half-disciplined on this and this is often not as cleaned up as it could (likely: should) be).